system/security/LdapSecurity/permissions.cpp

```diff
@@ -1322,18 +1328,20 @@ CSecurityDescriptor* PermissionProcessor::changePermission(CSecurityDescriptor*
         unsigned iAce;
 
         done = true;
-        for(iAce = 0; iAce < ASizeInfo.AceCount; iAce++)
+        for(iAce = 0; iAce < ASizeInfo.AceCount; iAce++)//For all ACE entries in the ACL
         {
-            GetAce(pdacl, iAce, (void**)&pAce);
+            GetAce(pdacl, iAce, (void**)&pAce);         //Get indexed ACE entry from ACL
             PSID cursid = (PSID)&(pAce->SidStart);
 
-            if(EqualSid(cursid, act_psid))
+            if(EqualSid(cursid, act_psid))              //Check for ACE to be replaced
             {
+#ifdef _DEBUG
                 if(stricmp(action.m_action.str(), "add") == 0)
-                    throw MakeStringException(-1, "Permission for account %s already exists", action.m_account_name.str());
-                DeleteAce(pdacl, iAce);
+                    DBGLOG("Permission for account %s already exists", action.m_account_name.str());
```

Comment:
When calling the 'add' action and the permission for account already exists, what will LDAP do (overwrite?, skip?, or ?) ?

Reply:
For an "add" the previous permission will be removed, and the new (added) permission will be granted. 
This is different than an "update:," where the permissions would be merged

Reply:
Got it